packages/requisitions/src/ResponseRequisition/api/api.ts

```diff
@@ -136,4 +136,23 @@ export const getResponseQueries = (sdk: Sdk, storeId: string) => ({
       return result.updateResponseRequisitionLine;
     } else throw new Error('Could not update response line');
   },
+  createOutboundFromResponse: async (responseId: string): Promise<number> => {
+    const result = await sdk.createOutboundFromResponse({
+      storeId,
+      responseId,
+    });
+
+    if (result.createRequisitionShipment.__typename === 'InvoiceNode') {
+      return result.createRequisitionShipment.invoiceNumber;
+    }
+
+    if (
+      result.createRequisitionShipment.__typename ===
+      'CreateRequisitionShipmentError'
+    ) {
+      throw new Error(result.createRequisitionShipment.error.__typename);
+    }
```

Comment:
I think this can be the start of a pattern.. maybe a little more refined though - but see the `onError` in the hook 🤔

Reply:
would be nice to make this generic and have a method that handles the response.. but I think the type handling would make it too painful

Reply:
🤔 

Not seeing how to make it nice and generic yet. I think it might be a bit of a long road of doing this all manually to see what can be abstracted